### PR TITLE
Build from source - add missing packages list

### PR DESCRIPTION
**Pull request by makr11st**
## Proposed changes

Build from source - add missing packages list

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/supported-platforms.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 Common requirements for all systems are:
 * minimum 16MB of RAM
-* systemd (for productive systems)
-* mosquitto minimum version 1.6 (for security reasons we recommend the latest 1.x version)
+* systemd (for production systems)
+* mosquitto minimum version 1.6 (for security reasons we recommend the latest 2.x version)
```
Comment (rina23q) [inline docs/src/supported-platforms.md:6]: Did we change the recommended mosquitto version to 2.x? I didn't know that :P

File: BUILDING.md [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+* git
+* Rust toolchain
+
+Following packages are required:
+
+* build-essentials
+* curl
+* gcc
+
+List of our test platforms can be found [here](docs/src/supported-platforms.md).
```
Comment (rina23q) [inline BUILDING.md:17]:
```suggestion
A list of our test platforms can be found [here](docs/src/supported-platforms.md).
```
advised by Grammerly

File: BUILDING.md [hunk 3]
```diff
@@ -1,5 +1,21 @@
 # Building thin-edge.io
 
+## Requirements
+
+You can use any OS to build from source (below has been tested on Ubuntu, but we also use Debian, macOS and FreeBSD successfully), our recommended setup and required tools:
```
Comment (rina23q) [inline BUILDING.md:5]: I think separating the sentence into two looks better.
```suggestion
You can use any OS to build from source (below has been tested on Ubuntu, but we also use Debian, macOS, and FreeBSD successfully). 

Our recommended setup and required tools are:
```

**Review by rina23q — APPROVED** [review]
Looks good. Only minor comments.

File: docs/src/supported-platforms.md (hunk 1 above)
Comment (makr11st) [reply docs/src/supported-platforms.md:6]: Good point, we didn't but the parenthesis recommendation kind didn't make much to me...

**Review by itsyitsy — APPROVED** [review]

**Pull request merged**